### PR TITLE
Add unit tests for normalization

### PR DESCRIPTION
**Pull request by tonywu1999**
### **User description**
# Motivation and Context

Please include relevant motivation and context of the problem along with a short summary of the solution.

## Changes

Please provide a detailed bullet point list of your changes.

- 

## Testing

Please describe any unit tests you added or modified to verify your changes.

## Checklist Before Requesting a Review
- [ ] I have read the MSstats [contributing guidelines](https://github.com/Vitek-Lab/MSstatsConvert/blob/master/.github/CONTRIBUTING.md)
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have run the devtools::document() command after my changes and committed the added files

___

### **PR Type**
Tests

___

### **Description**
- Add regression test for heavy-label retention

- Cover `MSstatsNormalize` no-op and median paths

- Validate `.normalizeMedian` with missing runs

- Test labeled-data shifts from heavy references

___

### Diagram Walkthrough

```mermaid
flowchart LR
  a["Label-based SRM data"]
  b["dataProcess preserves H rows"]
  c["Normalization test coverage"]
  d["Median, NA-run, and labeled cases"]
  a -- "validated by" --> b
  b -- "complements" --> c
  c -- "expanded into" --> d
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_dataProcess.R</strong><dd><code>Add heavy-label preservation regression test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

inst/tinytest/test_dataProcess.R

<ul><li>Add a regression test for label-based SRM data<br> <li> Assert <code>FeatureLevelData</code> retains heavy-label <code>H</code> rows<br> <li> Verify <code>dataProcess</code> preserves labeled experiment output</ul>

</details>

  </td>
  <td><a href="https://github.com/Vitek-Lab/MSstats/pull/188/files#diff-c9772a4fe72da4b2c35a77af03a328ab522a487807b5d2dd7e37ee3cde9cc3a5">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_utils_normalize.R</strong><dd><code>Expand median normalization unit test coverage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

inst/tinytest/test_utils_normalize.R

<ul><li>Add reusable helpers to build normalization inputs<br> <li> Test <code>MSstatsNormalize</code> for <code>NONE</code>, <code>FALSE</code>, and median normalization<br> <li> Verify <code>.normalizeMedian</code> equalizes run medians and removes temp columns<br> <li> Cover all-<code>NA</code> runs and heavy/labeled normalization shifts</ul>

</details>

  </td>
  <td><a href="https://github.com/Vitek-Lab/MSstats/pull/188/files#diff-1e2e0dcdf7bfa505e5177452356251c6b50ef76744cd6d191f1d995051d93ae5">+192/-1</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

## Motivation and Context

This PR increases test coverage for MSstats normalization and SRM data processing to prevent regressions and to validate edge-case behaviors (missing data, labeled SRM data).  
Solution summary: add regression test to ensure dataProcess preserves heavy-label ("H") rows for label-based SRM inputs, and add a comprehensive suite of unit tests for MSstatsNormalize and internal .normalizeMedian to validate NONE/FALSE/median behaviors, median-equalization, cleanup, NA-run handling, and heavy/light label shift behavior.

## Detailed Changes

- inst/tinytest/test_dataProcess.R
  - Added assertion that output from dataProcess(SRMRawData, use_log_file = FALSE) contains LABEL == "H" in QuantDataDefault$FeatureLevelData.
  - Integrated check into existing test flow prior to technical replicates/fractions tests.
  - Lines changed: +7/-0

- inst/tinytest/test_utils_normalize.R
  - Added comprehensive test suite for MSstatsNormalize and MSstats:::.normalizeMedian.
  - Introduced deterministic synthetic data.table helpers to construct normalization inputs.
  - Added tests covering:
    - MSstatsNormalize(..., "NONE") and MSstatsNormalize(..., "FALSE") return inputs unchanged.
    - MSstatsNormalize(..., "EQUALIZEMEDIANS") equalizes per-RUN medians while preserving row counts and within-run relative feature differences.
    - .normalizeMedian converges per-RUN medians to a common value and removes temporary columns ABUNDANCE_RUN and ABUNDANCE_FRACTION.
    - Normalization behavior when a RUN has all-NA ABUNDANCE: other runs remain correctly equalized and unaffected.
    - Labeled-data scenarios: normalization factors derived from heavy (H) rows equalize H medians, do not pre-equalize L medians, and apply identical abundance shifts to both L and H rows within the same run.
  - Fixed missing newline at EOF.
  - Lines changed: +192/-1

## Unit Tests Added or Modified

- test_dataProcess.R
  - One regression assertion added to validate heavy-label ("H") rows are preserved in processed SRM output.

- test_utils_normalize.R
  - New deterministic test helpers for generating synthetic normalization inputs.
  - Tests added to assert:
    - Identity behavior for NONE/FALSE modes (expect_identical).
    - Median equalization correctness and preservation of intra-run differences.
    - Correct convergence and removal of temporary columns by .normalizeMedian.
    - Robustness when a run contains only NA abundances.
    - Correct derivation and application of normalization shifts in mixed LABEL (L/H) runs, including that shifts computed from H are applied equally to both labels.

### CODE REVIEW

File: inst/tinytest/test_dataProcess.R [hunk 1]
```diff
@@ -15,6 +15,13 @@ expect_equal(nrow(QuantDataDefault$FeatureLevelData),
 expect_equal(nrow(QuantDataDefaultLinear$FeatureLevelData),
              nrow(QuantDataParallelLinear$FeatureLevelData))
 
+# SRMRawData is a label-based experiment: heavy ("H") rows must be preserved
+# in FeatureLevelData after dataProcess
+expect_true(
+    "H" %in% QuantDataDefault$FeatureLevelData$LABEL,
+    info = "FeatureLevelData should contain heavy-label (H) rows for label-based SRM data"
+)
```
[the remark this answers is not in the document]
Comment (tonywu1999) [reply inst/tinytest/test_dataProcess.R:23]: This test passes though.
